unyt/array.py

```diff
@@ -843,21 +843,14 @@ def in_units(self, units, equivalence=None, **kwargs):
                         RuntimeWarning,
                         stacklevel=2,
                     )
-            new_dtype = np.dtype("f" + str(dsize))
-            conversion_factor = new_dtype.type(conversion_factor)
-            ret = np.asarray(self.ndview * conversion_factor, dtype=new_dtype)
```

Comment:
Is there a reason this got switched from `ndview` to `v`? The latter involves making a copy, which we want to avoid if we can.

Reply:
Inadvertent change while prototyping the fix. I find that `ndview` works fine and will switch back to it.